# Algorithms for integration of single-cell transcriptomic data


Single-cell transcriptomics represents an innovative technique that allows for the examination of gene expression of individual cells in tissues or other complex biological samples. The data generated through this approach offer crucial insights into various biological domains, such as embryology, oncology, and immunology. Understanding the functioning and mutual interactions of individual cells within tissues indeed requires knowledge of their individual expression profiles. 

Given the novelty of this method and technical challenges faced during data generation, comparison of different samples may be difficult and affected by various batch effects. Effectively reconciling the information obtained regarding gene expression in distinct biological replicates or technical replicates generated by different methodologies may be onerous. The development of bioinformatics approaches to address this challenge is an ongoing process. 

The objective of this thesis is to:
1. survey the existing solutions for integration of single-cell transcriptomic data,
2. benchmark these solutions using publicly available datasets, specifically peripheral blood
mononuclear cells, and
3. select the most suitable integration method to apply to a tumour microenvironment dataset
generated by our laboratory.

This bachelor project encompasses the aforementioned steps, ultimately aiming to enhance our understanding of single-cell transcriptomics data integration.

## Compared tools 

| Programming Environment | Package / Method         | Type of Approach                          | Key Principle |
|--------------------------|--------------------------|-------------------------------------------|---------------|
| R                        | Seurat v5 (Integration)  | Graph-based / CCA / Anchoring             | Identifies mutual nearest neighbors (anchors) between datasets to define cell correspondences, and then corrects expression values based on these anchors. |
| R                        | Harmony                  | Iterative Clustering / Optimization       | Iteratively adjusts cell embeddings by maximizing diversity within clusters and minimizing batch effect across clusters using an objective function. |
| R                        | STACAS                  | Feature-level Correction                  | Aims to align expression profiles across batches using a shared feature space. |
| R                        | FastMNN                 | Mutual Nearest Neighbors                  | Corrects batch effects by identifying pairs of mutually nearest cells across different batches and computing a correction vector based on their difference. |
| Python                   | Scanorama               | Alignment / Nearest Neighbors             | Aligns individual batches by identifying "shared biological signal" across datasets through the detection of common gene features and subsequent correction. |
| Python                   | scVI                    | Variational Autoencoder (VAE) / Model-based | Uses a deep learning model to learn a low-dimensional, batch-corrected latent space for the data, providing a probabilistic framework for integration. |
| Python                   | CanSig                  | Consensus Clustering (Context-dependent)  | 	Specialized for Tumour scRNA-seq Integration (using scVI under the hood). It categorizes cells (malignant/non-malignant) and then employs a deep probabilistic generative model (from scVI) for integration, ensuring distinction between the two cell types is preserved. |

